fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchIncludeTest.java

```diff
@@ -0,0 +1,585 @@
+        procedure = procedure.toBuilder()
+                .status(ProcedureStatus.COMPLETED)
+                .subject(Reference.builder().reference(of("Patient/" + patient2Id)).build())
+                .basedOn(Reference.builder().reference(of("ServiceRequest/" + tag)).build())
+                .performed(DateTime.of(now.toString()))
+                .instantiatesUri(Uri.of("2" + tag))
+                .code(CodeableConcept.builder().coding(Coding.builder().code(Code.of("2" + tag)).build()).build())
+                .build();
+
+        // Call the 'create' API.
+        Entity<Procedure> entity = Entity.entity(procedure, FHIRMediaType.APPLICATION_FHIR_JSON);
+        Response response = target.path("Procedure").request().post(entity, Response.class);
+        assertResponse(response, Response.Status.CREATED.getStatusCode());
+
+        // Get the procedure's logical id value.
+        procedure2Id = getLocationLogicalId(response);
+
+        // Next, call the 'read' API to retrieve the new procedure and verify it.
+        response = target.path("Procedure/" + procedure2Id).request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+    }
+
+    @Test(groups = { "server-search-include" }, dependsOnMethods = {"testCreatePatient2"})
+    public void testCreateProcedure3() throws Exception {
+        WebTarget target = getWebTarget();
+
+        // Build a new Procedure and add subject reference to patient.
+        Procedure procedure = TestUtil.getMinimalResource(ResourceType.PROCEDURE, Format.JSON);
+        procedure = procedure.toBuilder()
+                .status(ProcedureStatus.COMPLETED)
+                .subject(Reference.builder().reference(of("Patient/" + patient2Id)).build())
+                .performed(DateTime.of(now.toString()))
+                .instantiatesUri(Uri.of("3" + tag))
+                .code(CodeableConcept.builder().coding(Coding.builder().code(Code.of("3" + tag)).build()).build())
+                .build();
+
+        // Call the 'create' API.
+        Entity<Procedure> entity = Entity.entity(procedure, FHIRMediaType.APPLICATION_FHIR_JSON);
+        Response response = target.path("Procedure").request().post(entity, Response.class);
+        assertResponse(response, Response.Status.CREATED.getStatusCode());
+
+        // Get the procedure's logical id value.
+        procedure3Id = getLocationLogicalId(response);
+
+        // Next, call the 'read' API to retrieve the new procedure and verify it.
+        response = target.path("Procedure/" + procedure3Id).request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+    }
+
+    @AfterClass
+    public void testDeleteResources() {
+        WebTarget target = getWebTarget();
+        if (patient1Id != null) {
+            Response response   = target.path("Patient/" + patient1Id).request(FHIRMediaType.APPLICATION_FHIR_JSON).delete();
```

Comment:
if this fails, the following entries are never executed, perhaps split them out?

Reply:
Done.